catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityFilterPlugin.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.metacard.validation;
+
+import static ddf.catalog.metacard.validation.MetacardValidityMarkerPlugin.VALIDATION_ERRORS;
+import static ddf.catalog.metacard.validation.MetacardValidityMarkerPlugin.VALIDATION_WARNINGS;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+
+public class MetacardValidityFilterPlugin implements PostQueryPlugin {
+
+    private static Map<String, List<String>> attributeMap = new HashMap<>();
+
+    @Override
+    public QueryResponse process(QueryResponse input)
+            throws PluginExecutionException, StopProcessingException {
+        List<Result> results = input.getResults();
+
+        // process security
+        for (Result result : results) {
+            Metacard metacard = result.getMetacard();
+            Attribute securityAttribute = metacard.getAttribute(Metacard.SECURITY);
+            HashMap<String, List<String>> securityMap;
+            if (securityAttribute != null && securityAttribute.getValue() != null
+                    && securityAttribute.getValue() instanceof Map) {
+                securityMap = (HashMap<String, List<String>>) securityAttribute.getValue();
```

Comment:
Unfortunately this ain't doing what you expect it too. Because of type erasure and stuff you can't instanceof a generic. The only way to really check is do the cast and catch the runtime ClassCastException (since generics are a runtime detail). So you'll still want to try catch that here.